### ISSUE TITLE
Callback warning error

### ISSUE THREAD

**Issue by psinger** [keras-team/keras#12069]
I receive the following error when using `fit_generator` with callbacks in Keras:

```
/opt/conda/lib/python3.6/site-packages/Keras-2.2.4-py3.6.egg/keras/callbacks.py in _call_batch_hook(self, mode, hook, batch, logs)
     93                 'Method (%s) is slow compared '
     94                 'to the batch update (%f). Check your callbacks.', hook_name,
---> 95                 delta_t_median)
     96         if hook == 'begin':
     97             self._t_enter_batch = time.time()

TypeError: integer argument expected, got float

```

Apparently it happens with some combination of warnings package and Keras packages, or something else. Any idea for a quick workaround that does not involve removing callbacks?

**Comment by gabrieldemarmiesse** [comment]
This is because the formatting is a bit too strict on the types. It's an easy fix. Contribution welcome. Thanks for the report!

**Comment by Abelarm** [comment]
Hi Gabriel,
I'd like to work on this.

**Comment by gabrieldemarmiesse** [comment]
Solved by #12098

**Comment by robotdreams** [comment]
Hello, 

I have the same problem at Kaggle when implementing cyclical learning rate using the Callback Class.  And it is very strange for me, because the code works only with VGG16. But with wother networks I get the message:

Epoch 1/10
 2/91 [..............................] - ETA: 29:56 - loss: 0.9238 - acc: 0.4500
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-21-507d26ad5ed3> in <module>()
     26                               validation_steps = len(val_list) // batch_size + 1,
     27                               callbacks=[clr],
---> 28                               verbose=1)

/opt/conda/lib/python3.6/site-packages/Keras-2.2.4-py3.6.egg/keras/legacy/interfaces.py in wrapper(*args, **kwargs)
     89                 warnings.warn('Update your `' + object_name + '` call to the ' +
     90                               'Keras 2 API: ' + signature, stacklevel=2)
---> 91             return func(*args, **kwargs)
     92         wrapper._original_function = func
     93         return wrapper

/opt/conda/lib/python3.6/site-packages/Keras-2.2.4-py3.6.egg/keras/engine/training.py in fit_generator(self, generator, steps_per_epoch, epochs, verbose, callbacks, validation_data, validation_steps, class_weight, max_queue_size, workers, use_multiprocessing, shuffle, initial_epoch)
   1434             use_multiprocessing=use_multiprocessing,
   1435             shuffle=shuffle,
-> 1436             initial_epoch=initial_epoch)
   1437 
   1438     @interfaces.legacy_generator_methods_support

/opt/conda/lib/python3.6/site-packages/Keras-2.2.4-py3.6.egg/keras/engine/training_generator.py in fit_generator(model, generator, steps_per_epoch, epochs, verbose, callbacks, validation_data, validation_steps, class_weight, max_queue_size, workers, use_multiprocessing, shuffle, initial_epoch)
    217                     batch_logs[l] = o
    218 
--> 219                 callbacks._call_batch_hook('train', 'end', batch_index, batch_logs)
    220 
    221                 batch_index += 1

/opt/conda/lib/python3.6/site-packages/Keras-2.2.4-py3.6.egg/keras/callbacks.py in _call_batch_hook(self, mode, hook, batch, logs)
     93                 'Method (%s) is slow compared '
     94                 'to the batch update (%f). Check your callbacks.', hook_name,
---> 95                 delta_t_median)
     96         if hook == 'begin':
     97             self._t_enter_batch = time.time()

TypeError: integer argument expected, got float

Please help me to solve this problem.
Best
